packages/blade/src/components/Input/BaseInput/baseInputStyles.ts

```diff
@@ -0,0 +1,111 @@
+import type { CSSObject } from 'styled-components';
+import type { BaseInputProps } from './BaseInput';
+import { getInputVisualsToBeRendered } from './BaseInputVisuals';
+import type { Theme } from '~components/BladeProvider';
+import getTextStyles from '~components/Typography/Text/getTextStyles';
+import { getPlatformType, makeBorderSize, makeSpace } from '~utils';
+
+type GetInputStyles = Pick<
+  BaseInputProps,
+  | 'isDisabled'
+  | 'validationState'
+  | 'leadingIcon'
+  | 'prefix'
+  | 'interactionElement'
+  | 'suffix'
+  | 'trailingIcon'
+> & {
+  isFocused?: boolean;
+  theme: Theme;
+};
+
+export const getInputBackgroundAndBorderStyles = ({
+  theme,
+  isFocused,
+  isDisabled,
+  validationState,
+}: GetInputStyles): CSSObject => {
```

Comment:
Let's not set a wider type in the parameter when we are actually using just theme isFocused, isDisabled and validationState. Will cause confusion while using the function.

Reply:
didn't get you?

Reply:
`GetInputStyles` have everything including: 

```
  | 'isDisabled'
  | 'validationState'
  | 'leadingIcon'
  | 'prefix'
  | 'interactionElement'
  | 'suffix'
  | 'trailingIcon'
   & {
  isFocused?: boolean;
  theme: Theme;
};
```

But in this function we are not using all of them but still adding that type. 

It's somewhat misleading: 

<img width="542" alt="image" src="https://user-images.githubusercontent.com/35374649/186349915-19a4ab5d-1627-4fd1-be8d-08f9938d1c0d.png">

Reply:
fixed in e263fd9b727b4b26f7de680d3ce34de705cca296